File: src/main/java/pl/mroz/buddiesapi/domain/rental/RentalDomainService.java
package pl.mroz.buddiesapi.domain.rental;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;
import java.util.UUID;

public interface RentalDomainService {

    List<Rental> getAllRentals();

    Page<Rental> getByCriteria(RentalRepository.IRentalCriteria rentalCriteria, Pageable pageable);

    List<Rental> getRentalsFromUser(UUID accountId);

    Rental getRental(UUID rentalId);

    List<String> getTopTags();

    Rental createRental(Rental rental);

    Rental updateRental(UUID uuid, Rental rental);

    void deleteRental(Rental rental);
}

File: src/main/kotlin/buddiesapi/domain/rental/RentalDomainService.kt
package buddiesapi.domain.rental

import java.util.UUID
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

interface RentalDomainService {
    fun getAllRentals(): List<Rental>

    fun getByCriteria(rentalCriteria: RentalRepository.IRentalCriteria, pageable: Pageable): Page<Rental>

    fun getRentalsFromUser(accountId: UUID): List<Rental>

    fun getRental(rentalId: UUID): Rental

    fun getTopTags(): List<String>

    fun createRental(rental: Rental): Rental

    fun updateRental(uuid: UUID, rental: Rental): Rental

    fun deleteRental(rental: Rental)
}
